synapse/api/errors.py

```diff
@@ -70,17 +74,44 @@ def __init__(self, code, msg, errcode=Codes.UNKNOWN):
         Args:
             code (int): The integer error code (an HTTP response code)
             msg (str): The human-readable error message.
-            err (str): The error code e.g 'M_FORBIDDEN'
+            errcode (str): The synapse error code e.g 'M_FORBIDDEN'
         """
-        super(SynapseError, self).__init__(code, msg)
+        super(SynapseError, self).__init__(code)
+        self.msg = msg
         self.errcode = errcode
 
+    def __str__(self):
+        return "%d: %s %s" % (self.code, self.errcode, self.msg)
+
     def error_dict(self):
         return cs_error(
             self.msg,
             self.errcode,
         )
 
+    @classmethod
+    def from_http_response_exception(cls, err):
+        """Make a SynapseError based on an HTTPResponseException
```

Comment:
Please document how this converts between the two. It's not clear to me how this conversion should take place. Does a 400 get proxied through? A 401? Or does it just allow through well known OK error codes such as 404?